compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -437,90 +439,96 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
 
         debug!("report_region_errors: {} errors after preprocessing", errors.len());
 
-        for error in errors {
-            debug!("report_region_errors: error = {:?}", error);
-
-            if !self.try_report_nice_region_error(&error) {
-                match error.clone() {
-                    // These errors could indicate all manner of different
-                    // problems with many different solutions. Rather
-                    // than generate a "one size fits all" error, what we
-                    // attempt to do is go through a number of specific
-                    // scenarios and try to find the best way to present
-                    // the error. If all of these fails, we fall back to a rather
-                    // general bit of code that displays the error information
-                    RegionResolutionError::ConcreteFailure(origin, sub, sup) => {
-                        if sub.is_placeholder() || sup.is_placeholder() {
-                            self.report_placeholder_failure(origin, sub, sup).emit();
-                        } else {
-                            self.report_concrete_failure(origin, sub, sup).emit();
+        let guars: Vec<_> = errors
```

Comment:
instead of using a `Vec` :thinking: maybe just use an `Option` and overwrite it for each goal?

Reply:
ok